### ISSUE TITLE
reportArgumentType when passing `type` annotated parameter to `super()`

### ISSUE THREAD

**Issue by peterschutt** [microsoft/pyright#7931]
**Describe the bug**
Calling `super` with a parameter annotated as `type` such as:

```python
def fn(t: type) -> super:
    return super(t, t)


class MyClass:
    pass


fn(MyClass)
```

Results in:

```
test.py:3:18 - error: Expected class type as first argument to "super" call but received "type" (reportArgumentType)
1 error, 0 warnings, 0 informations 
```

**VS Code extension or command-line**
CLI

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.364.

**Comment by peterschutt** [comment]
Thank you!
